app/src/main/java/me/saket/dank/widgets/IndentedLayout.java

```diff
@@ -10,41 +10,81 @@
 import androidx.annotation.IntRange;
 import androidx.annotation.Nullable;
 import androidx.annotation.Px;
+import com.f2prateek.rx.preferences2.Preference;
+import io.reactivex.Observable;
+
 import android.util.AttributeSet;
 import android.util.TypedValue;
 import android.widget.LinearLayout;
 
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
 import me.saket.dank.R;
+import me.saket.dank.di.Dank;
 
 public class IndentedLayout extends LinearLayout {
-
-  private static final int DEFAULT_SPACING_PER_DEPTH_DP = 8;
-  private static final int DEFAULT_LINE_WIDTH = 4;
+  private static final int DEFAULT_SPACING_PER_DEPTH_DP = 10;
+  private static final int DEFAULT_LINE_WIDTH = 6;
 
   private int indentationDepth;
   private final int spacePerDepthPx;
   private Paint indentationLinePaint;
-  private Path indentationLinePath;
+  private int indentationLineWidth;
+  private int defaultIndentationLineColor;
 
   private int originalPaddingStart;
 
+  private static final String[] INDENTATION_COLORS = {
+      "#FFC40D", // Yellow
+      "#2D89EF", // Blue
+      "#B91D47", // Dark Red
+      "#00ABA9", // Teal
+      "#E3A21A", // Orange
+      "#99B433", // Light Green
+      "#7E3878", // Purple
+      "#FFB300", // Vivid Yellow
+      "#FFFFFF", // White
+      "#00A300", // Green
+      "#2B5797", // Dark Blue
+      "#9F00A7", // Light Purple
+      "#603CBA", // Dark Purple
+      "#EE1111", // Red
+      "#EFF4FF", // Light Blue
+      "#DA532C", // Dark Orange
+      "#FF0097", // Magenta
+      "#1E7145", // Dark Green
+  };
+
+  private List<ColoredTree> trees = new ArrayList<>();
+  @Inject @Named("show_colored_comments_tree") Preference<Boolean> coloredDepthPreference;
+
   public IndentedLayout(Context context, @Nullable AttributeSet attrs) {
     super(context, attrs);
     setWillNotDraw(false);
 
+    Dank.dependencyInjector().inject(this);
+
+    coloredDepthPreference.asObservable()
+        .subscribe(o -> {
+          this.coloredDepthPreference.set(o.booleanValue());
```

Comment:
`coloredDepthPreference` already stores the value received by this subscriber. Rx preferences are probably done to avoid notifying if the setting doesn't change but if it weren't for that this line would cause an infinite loop.

Instead you could simply call `setupDepthLines` to update the lines. That method should then correctly read new setting by itself. Call to `invalidate` should be kept though.